mypyc/irbuild/ll_builder.py

```diff
@@ -649,20 +653,23 @@ def add_bool_branch(self, value: Value, true: BasicBlock, false: BasicBlock) ->
         self.add(Branch(value, true, false, Branch.BOOL_EXPR))
 
     def call_c(self,
-               function_name: str,
+               desc: CFunctionDescription,
                args: List[Value],
-               line: int,
-               result_type: Optional[RType]) -> Value:
+               line: int) -> Value:
         # handle void function via singleton RVoid instance
-        ret_type = void_rtype if result_type is None else result_type
-        target = self.add(CallC(function_name, args, ret_type, line))
+        ret_type = void_rtype if desc.result_type is None else desc.result_type
+        coerced = []
+        for i, arg in enumerate(args):
+            formal_type = desc.arg_types[i]
+            arg = self.coerce(arg, formal_type, line)
+            coerced.append(arg)
+        target = self.add(CallC(desc.c_function_name, coerced, ret_type, desc.steals, line))
```

Comment:
Also coerce the result into `result_type`? Maybe rename `desc.result_type` to `desc.return_type` or something to avoid confusion. Return type would be the type of the value returned by the function, and result type would be the type that the return value needs to be coerced to, based on the context.

Reply:
This is super reasonable and I clearly miss that distinction.